Commands/notify.py

```diff
@@ -0,0 +1,267 @@
+                    if reaction_emoji == v["emoji"]:
+
+                        msg_string = await guild_data.sub_user(key, user.id)
+                        await ctx.send(msg_string)
+
+            except asyncio.TimeoutError:
+                pass
+
+            if time.time() > timeout:
+                break
+
+    async def wait_for_removed_reactions(self, ctx, msg, guild_data, timeout):
+        while True:
+            try:
+                reaction, user = await ctx.bot.wait_for(
+                    "reaction_remove",
+                    check=lambda reaction, user: reaction.message.id == msg.id
+                    and not user.bot,
+                    timeout=30.0,
+                )
+                if reaction.custom_emoji:
+                    reaction_emoji = str(reaction.emoji.id)
+                else:
+                    reaction_emoji = reaction.emoji
+                for key, v in guild_data.notification_lists.items():
+
+                    if reaction_emoji == v["emoji"]:
+
+                        msg_string = await guild_data.unsub_user(key, user.id)
+                        await ctx.send(msg_string)
+
+            except asyncio.TimeoutError:
+                pass
+
+            if time.time() > timeout:
+                break
+
+    @commands.command(name="show_lists")
+    async def show_lists(self, ctx):
+        # TODO printing of custom emojis
+        guild_data = await get_guild_data(ctx.message.guild.id)
+
+        if guild_data.notification_lists:
+            text = "Lists:\n"
+            for k, v in guild_data.notification_lists.items():
+                if v["is_custom_emoji"]:
+                    # TODO: Extract to 'get_custom_emoji' method for reusability
+                    text += ("\n<:"
+                             + ctx.bot.get_emoji(int(v["emoji"])).name
+                             + ":"
+                             + v["emoji"]
+                             + "> - "
+                             + k
+                             )
+                else:
+                    text += "\n" + v["emoji"] + " - " + k
+            # text += "\n".join(
+            #     [
+            #         v["emoji"] + " - " + k
+            #         for k, v in guild_data.notification_lists.items()
+            #     ]
+            # )
+
+            msg = await ctx.send(text)
+            for v in guild_data.notification_lists.values():
+                await msg.add_reaction(v["emoji"] if not v["is_custom_emoji"] else ctx.bot.get_emoji(int(v["emoji"])))
+
+            # TODO make reaction time configurable
+            timeout = time.time() + 60  # 1 minutes from now
+            reaction_added_task = asyncio.create_task(
+                self.wait_for_added_reactions(ctx, msg, guild_data, timeout)
+            )
+            reaction_removed_task = asyncio.create_task(
+                self.wait_for_removed_reactions(ctx, msg, guild_data, timeout)
+            )
+
+            await reaction_added_task
```

Comment:
I'm going to need an explanation on how this works exactly.
What is it that both tasks do exactly?

Reply:
There is one task that waits for reactions to be added and then adds the user to the list, and another one that waits for a reaction to be removed and then removes the user from the list. 

It is necessary to make them separate tasks so that both the added and removed reactions can be monitored simultaneously.